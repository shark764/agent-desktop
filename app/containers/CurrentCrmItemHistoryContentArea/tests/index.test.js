/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import { CurrentCrmItemHistoryContentArea } from '../index';

describe('<CurrentCrmItemHistoryContentArea />', () => {
  it('should render correctly', () => {
    const rendered = shallow(
      <CurrentCrmItemHistoryContentArea
        zendeskActiveTab={fromJS({
          id: 123,
          type: 'user',
          attributes: {
            name: 'test-name',
          },
        })}
        closeCurrentCrmItemHistoryPanel={() => {}}
        loadCrmInteractionHistory={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
