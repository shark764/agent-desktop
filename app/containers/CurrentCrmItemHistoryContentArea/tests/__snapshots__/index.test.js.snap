// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentCrmItemHistoryContentArea /> should render correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "padding": "20px 0 0 20px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexShrink": 0,
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "marginRight": "5px",
        }
      }
    >
      <div
        style={
          Object {
            "marginBottom": "5px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Interaction history for:"
          id="app.containers.CurrentCrmItemHistoryContentArea.interactionHistoryFor"
          values={Object {}}
        />
      </div>
      <div
        className="crmItemName"
        title="test-name"
      >
        <Dotdotdot
          clamp={3}
          truncationChar="â€¦"
          useNativeClamp={true}
        >
          test-name
        </Dotdotdot>
      </div>
    </div>
    <InjectIntl(Button)
      id="cancelCurrentCrmItemHistoryButton"
      onClick={[Function]}
      style={
        Object {
          "float": "right",
          "height": "35px",
          "marginRight": "20px",
        }
      }
      text={
        Object {
          "defaultMessage": "Cancel",
          "id": "app.containers.CurrentCrmItemHistoryContentArea.cancel",
        }
      }
      type="secondary"
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexGrow": 1,
      }
    }
  >
    <Connect(ErrorBoundaryComponent)
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </div>
</div>
`;
